interfaces/builtin/adb.go

```diff
@@ -0,0 +1,226 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+)
+
+const adbSummary = `allows operating as Android Debug Bridge service`
+
+const adbBaseDeclarationSlots = `
+  adb:
+    allow-installation:
+      slot-snap-type:
+        - app
+    deny-auto-connection: true
+`
+
+var adbVendorList = map[int]string{
```

Comment:
Where does this list come from, can you link to a source if any?

Reply:
It was derived from this post: https://forum.snapcraft.io/t/udev-rules-for-fastboot-and-adb/5443/3

I also added the BQ vendor ID for the purpose of my Aquaris phone.

Reply:
Ah, I see, thanks, I was hoping there was an authoritative list somewhere. In any case, might be worth linking that forum post as a comment in the code.

Reply:
If that would help, we could create and maintain an authoritative list of rules in the name of @ubports.